sat/metrics/otel_metrics.go

```diff
@@ -32,7 +32,7 @@ func setupOtelMetrics(ctx context.Context, config options.MetricsConfig) (Metric
 		return Metrics{}, errors.Wrap(err, "otel metrics grpc connection")
 	}
 
-	err = observability.OtelMeter(ctx, conn, observability.MeterConfig{CollectPeriod: otelCollectionPeriod})
+	_, err = observability.OtelMeter(ctx, conn, observability.MeterConfig{CollectPeriod: otelCollectionPeriod})
```

Comment:
Yes, the shutdown function is important. Ideally it should be a case of

```go
meterShutdown, err := observability.OtelMeter(...)
if err != nil { ...}

defer meterShutdown()
```

But that might also mean that the meter needs to be configured from the main function, otherwise the meter shuts down once configuration of it is done, but before sat does anything else.

Reply:
It needs a refactor, yes.

Reply:
We need that now. What I would do is return the shutdown function / err from this function to main, export `configureMetrics`, and call `metrics.ConfigureMetrics` from main, if setupOtelMetrics did not error.

ConfigureMetrics depends on the meter being set up previously, but it's stored in a global, which configure metrics grabs it from. By default the global meter is a noop one.